src/Datadog.Trace/Tracer.cs

```diff
@@ -667,12 +667,19 @@ private static IDogStatsd CreateDogStatsdClient(TracerSettings settings, string
                 }
 
                 var statsd = new DogStatsdService();
-                statsd.Configure(new StatsdConfig
+                if (AzureAppServices.Metadata.IsRelevant)
```

Comment:
Instead of checking for AAS, should we check if named pipes are enabled (regardless of environment)?

Reply:
I thought about this, and it's subject to change in the future.
Right now, named pipes usage for dogstatsd is a combination the pipe variable being set, and the absence of the agent host variable.
I'd like to work with Olivier to make this more explicit for stats.

Reply:
I agree it should be fixed in the dogstatsd client, but right now, if I wanted to enable named pipes on my machine (outside of AAS), wouldn't I still have the issue you are trying fixing?

Reply:
If you wanted runtime metrics, and also named pipes on your local machine, you would need to currently set `DD_AZURE_APP_SERVICES:true` plus all of the named pipes variables.
The named pipe is an undocumented and not officially supported feature, so I don't see it being used outside of AAS until we explicitly decide to expand support for it.

Reply:
I see. Thanks for clarifying! I'll resolve this thread.